File: RepoTransBench/target_projects/Java/Go/tuenti_SmsRadar/tests/original/sms_test.go
package original

import (
	"testing"
	"github.com/stretchr/testify/assert"
)

func TestSms_ConstructorAndGetters(t *testing.T) {
	sms := NewSms("12345", "1687221000000", "Hello there!", RECEIVED)
	assert.Equal(t, "12345", sms.GetAddress())
	assert.Equal(t, "1687221000000", sms.GetDate())
	assert.Equal(t, "Hello there!", sms.GetMsg())
	assert.Equal(t, RECEIVED, sms.GetType())
}

func TestSms_EqualsHashCodeAndToString(t *testing.T) {
	a := NewSms("a", "111", "body", RECEIVED)
	b := NewSms("a", "111", "body", RECEIVED)
	c := NewSms("b", "112", "other", SENT)

	assert.True(t, a.Equal(b))
	assert.True(t, b.Equal(a))
	assert.False(t, a.Equal(c))
	assert.Contains(t, a.Msg, "body")
}

func TestSms_NotEqualWithNullOrOtherType(t *testing.T) {
	sms := NewSms("a", "c", "b", UNKNOWN)
	assert.False(t, sms.Equal(Sms{}))           // not equal to zero value unless all fields equal
}

File: RepoTransBench/source_projects/Java/tuenti_SmsRadar/library/src/test/java/com/tuenti/smsradar/SmsTest.java
package com.tuenti.smsradar;

import org.junit.Test;
import static org.junit.Assert.*;

public class SmsTest {

    @Test
    public void testConstructorAndGetters() {
        Sms sms = new Sms("12345", "1687221000000", "Hello there!", SmsType.RECEIVED);
        assertEquals("12345", sms.getAddress());
        assertEquals("1687221000000", sms.getDate());
        assertEquals("Hello there!", sms.getMsg());
        assertEquals(SmsType.RECEIVED, sms.getType());
    }

    @Test
    public void testEqualsHashCodeAndToString() {
        Sms a = new Sms("a", "111", "body", SmsType.RECEIVED);
        Sms b = new Sms("a", "111", "body", SmsType.RECEIVED);
        Sms c = new Sms("b", "112", "other", SmsType.SENT);

        assertEquals(a, b);
        assertEquals(a.hashCode(), b.hashCode());
        assertNotEquals(a, c);
        assertNotEquals(a.hashCode(), c.hashCode());
        assertTrue(a.toString().contains("body"));
    }

    @Test
    public void testNotEqualWithNullOrOtherType() {
        Sms sms = new Sms("a", "c", "b", SmsType.UNKNOWN);
        assertNotEquals(null, sms);
        assertNotEquals("Some String", sms);
    }
}